public/pages/workflows/new_workflow/utils.ts

```diff
@@ -138,68 +145,136 @@ export function fetchEmptyUIConfig(): WorkflowConfig {
   };
 }
 
-export function fetchSemanticSearchMetadata(): UIState {
+export function fetchSemanticSearchMetadata(version: string): UIState {
+  if (!version) {
+    throw new Error(
+      'Version parameter is required for fetchSemanticSearchMetadata'
+    );
+  }
   let baseState = fetchEmptyMetadata();
   baseState.type = WORKFLOW_TYPE.SEMANTIC_SEARCH;
-  baseState.config.ingest.enrich.processors = [new MLIngestProcessor().toObj()];
+  const isPreV219 = semver.lt(version, MINIMUM_FULL_SUPPORTED_VERSION);
+
+  baseState.config.ingest.enrich.processors = isPreV219
+    ? [
+        new TextEmbeddingIngestProcessor().toObj(),
+        new TextImageEmbeddingIngestProcessor().toObj(),
+      ]
+    : [new MLIngestProcessor().toObj()];
+
   baseState.config.ingest.index.name.value = generateId('knn_index', 6);
   baseState.config.ingest.index.settings.value = customStringify({
     [`index.knn`]: true,
   });
-  baseState.config.search.request.value = customStringify(TERM_QUERY_TEXT);
-  baseState.config.search.enrichRequest.processors = [
-    injectQueryTemplateInProcessor(
-      new MLSearchRequestProcessor().toObj(),
-      KNN_QUERY
-    ),
-  ];
+
+  baseState.config.search.request.value = customStringify(
+    isPreV219 ? SEMANTIC_SEARCH_QUERY_NEURAL : KNN_QUERY
+  );
+
+  baseState.config.search.enrichRequest.processors = isPreV219
+    ? []
+    : [
+        injectQueryTemplateInProcessor(
+          new MLSearchRequestProcessor().toObj(),
+          KNN_QUERY
+        ),
+      ];
+
   return baseState;
 }
 
-export function fetchMultimodalSearchMetadata(): UIState {
+export function fetchMultimodalSearchMetadata(version: string): UIState {
+  if (!version) {
+    throw new Error(
+      'Version parameter is required for fetchSemanticSearchMetadata'
+    );
+  }
+
   let baseState = fetchEmptyMetadata();
   baseState.type = WORKFLOW_TYPE.MULTIMODAL_SEARCH;
-  baseState.config.ingest.enrich.processors = [new MLIngestProcessor().toObj()];
+
+  const isPreV219 = semver.lt(version, MINIMUM_FULL_SUPPORTED_VERSION);
+
+  baseState.config.ingest.enrich.processors = isPreV219
+    ? [
+        new TextEmbeddingIngestProcessor().toObj(),
+        new TextImageEmbeddingIngestProcessor().toObj(),
+      ]
```

Comment:
Should just be the text image processor - see [multimodal docs](https://opensearch.org/docs/latest/search-plugins/multimodal-search/#step-1-create-an-ingest-pipeline) for reference.

Reply:
addressed on line 199